### PR TITLE
Scheduler — Popup keep resource state between openings

### PR DESCRIPTION
**Pull request by aleksei-semikozov**

### CODE REVIEW

File: packages/devextreme/js/__internal/scheduler/appointment_popup/m_popup.ts [hunk 1]
```diff
@@ -69,10 +69,10 @@ export class AppointmentPopup {
...
-    if (!this._popup) {
-      const popupConfig = this._createPopupConfig();
-      this._createPopup(popupConfig);
-    }
+    this._disposePopup();
+
+    const popupConfig = this._createPopupConfig();
+    this._createPopup(popupConfig);
 
     this._popup!.show();
```
[the remark this answers is not in the document]
Comment (aleksei-semikozov) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/m_popup.ts:77]: This is an edge case — showAppointmentPopup() cannot be triggered while the modal popup is already visible

File: packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts [hunk 2]
```diff
@@ -939,36 +953,6 @@ describe('Appointment Form', () => {
       expect(POM.popup.getInputValue('endDateEditor')).toBe('5/1/2017');
       expect(POM.popup.isInputVisible('endTimeEditor')).toBeFalsy();
     });
```
[the remark this answers is not in the document]
Comment (aleksei-semikozov) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:880]: These tests validated state reset within a reused form instance.

File: packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts [hunk 3]
```diff
@@ -441,6 +441,20 @@ describe('Appointment Form', () => {
...
+
+      scheduler.showAppointmentPopup(commonAppointment);
+      const firstFormInstance = POM.popup.dxForm;
+
+      POM.popup.cancelButton.click();
+
+      scheduler.showAppointmentPopup(commonAppointment);
+      const secondFormInstance = POM.popup.dxForm;
+
+      expect(secondFormInstance).not.toBe(firstFormInstance);
```
[the remark this answers is not in the document]
Comment (aleksei-semikozov) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:391]: Intentional. The behavioral test already exists above (lines 430-441, verifying resource state doesn't leak). This test explicitly validates the implementation contract

File: packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts [hunk 4]
```diff
@@ -939,36 +953,6 @@ describe('Appointment Form', () => {
...
-      POM.popup.getInput('allDayEditor').click();
-      POM.popup.cancelButton.click();
-
-      scheduler.showAppointmentPopup(commonAppointment);
-
-      expect(POM.popup.getInputValue('startDateEditor')).toBe('5/9/2017');
-      expect(POM.popup.getInputValue('startTimeEditor')).toBe('9:30 AM');
-      expect(POM.popup.getInputValue('endDateEditor')).toBe('5/9/2017');
-      expect(POM.popup.getInputValue('endTimeEditor')).toBe('11:00 AM');
-    });
```
Comment (Tucchhaa) [inline packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:971]: I think there are more state tests that can be removed. E.g.: https://github.com/DevExpress/DevExtreme/blob/d11078bb54f09723e0efdf0c719e069b47b3a6f6/packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts#L380

Can you pls check what kind of tests are not needed anymore? Maybe also testcafe tests

Comment (aleksei-semikozov) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:971]: Yes remove more 4 state tests jest test.
Checked testcafe tests for state isolation patterns (close popup → reopen popup). The key indicator is tests where the popup is opened twice within a single test case.
New form (appointmentForm/): No tests open the popup twice.
Legacy form (legacyAppointmentForm/): Found 3 state isolation tests(but wont fixed this unused code)

Comment (Tucchhaa) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:971]: Remove this test too pls:
https://github.com/DevExpress/DevExtreme/blob/0439dadc81b4c95e1928c4c2cab8a316a3c62335/packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts#L1531

Comment (aleksei-semikozov) [reply packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts:971]: removed it

**Review by Tucchhaa — APPROVED** [review]

**Review by sjbur — APPROVED** [review]
